### PR TITLE
feat: create e2e tests for organization custom roles page

### PR DESCRIPTION
**Pull request by jaaydenh**
Adds 5 premium tests and 1 non-premium test.

### CODE REVIEW

File: site/e2e/helpers.ts [hunk 1]
```diff
@@ -48,6 +48,10 @@ export function requiresLicense() {
 	test.skip(!license);
 }
 
+export function noPremiumLicense() {
```
Comment (aslilac) [inline site/e2e/helpers.ts:51]:
```suggestion
export function requiresUnlicensed() {
```

might jive a bit better with the existing `requiresLicense` helper

File: site/e2e/tests/deployment/idpOrgSync.spec.ts [hunk 2]
```diff
@@ -117,7 +117,7 @@ test.describe("IdpOrgSyncPage", () => {
 		requiresLicense();
 		await setupApiCalls(page);
 
-		await createOrganizationWithName("developers");
+		await createOrganizationWithName("admins");
```
Comment (aslilac) [inline site/e2e/tests/deployment/idpOrgSync.spec.ts:120]:
```suggestion
		await createOrganizationWithName(randomName());
```

avoid fixed names in e2e tests

File: site/e2e/tests/organizations/customRoles/customRoles.spec.ts [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+		// Verify that the more menu (three dots) is not present for built-in roles
+		await expect(
+			roleRow.getByRole("button", { name: "More options" }),
+		).not.toBeVisible();
+
+		await deleteOrganization("testers");
+	});
+
+	test("create custom role with UI", async ({ page }) => {
```
Comment (aslilac) [inline site/e2e/tests/organizations/customRoles/customRoles.spec.ts:120]: why not make this a helper and call it from the other test rather than using the api?

**Review by aslilac — COMMENTED** [review]
I was expecting assigning custom roles to be tested as part of this work, but we can do that in a separate PR if you'd like. It should basically be a copy+paste of the existing test for regular roles.

File: site/e2e/tests/organizations/customRoles/customRoles.spec.ts (hunk 3 above)
Comment (jaaydenh) [reply site/e2e/tests/organizations/customRoles/customRoles.spec.ts:120]: I have actually been going the route of having 2 different types of tests, one that starts with a change using the api and then completing the test through the UI and then another that does the same thing through the UI. I think there is the potential catching edge cases through this way of testing.

Comment (jaaydenh) [reply site/e2e/tests/organizations/customRoles/customRoles.spec.ts:124]: @aslilac Looking around at many of the existing tests, I see this pattern when the same things to happen for many tests, for example creating a template, the API is used but then atleast one test would do the same thing through the UI if possible. I feel like this many be a bit more performant than going through the UI for every test. The except would be if there are many variations of path the user could take through the UI.

**Review by aslilac — APPROVED** [review]
